synapseclient/api/entity_services.py

```diff
@@ -0,0 +1,173 @@
+        synapse_client: If not passed in or None this will use the last client from
+            the `.login()` method.
+
+    Returns:
+        The requested entity bundle matching
+            <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/Entity.html>
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+    return await client.rest_get_async(
+        uri=f"/entity/{entity_id}",
+    )
+
+
+@alru_cache(ttl=60)
+async def get_upload_destination(
+    entity_id: str, synapse_client: Optional["Synapse"] = None
+) -> Dict[str, Union[str, int]]:
+    """
+    <https://rest-docs.synapse.org/rest/GET/entity/id/uploadDestination.html>
+
+    Arguments:
+        entity_id: The ID of the entity.
+        synapse_client: If not passed in or None this will use the last client from
+            the `.login()` method.
+
+    Returns:
+        The upload destination.
+        <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/file/UploadDestination.html>
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+    return await client.rest_get_async(
+        uri=f"/entity/{entity_id}/uploadDestination",
+        endpoint=client.fileHandleEndpoint,
+    )
+
+
+async def get_upload_destination_location(
+    entity_id: str, location: str, synapse_client: Optional["Synapse"] = None
+) -> Dict[str, Union[str, int]]:
+    """
+    <https://rest-docs.synapse.org/rest/GET/entity/id/uploadDestination/storageLocationId.html>
+
+    Arguments:
+        entity_id: The ID of the entity.
+        location: A storage location ID of the upload destination.
+        synapse_client: If not passed in or None this will use the last client from
+            the `.login()` method.
+
+    Returns:
+        The upload destination.
+        <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/file/UploadDestination.html>
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+    return await client.rest_get_async(
+        uri=f"/entity/{entity_id}/uploadDestination/{location}",
+        endpoint=client.fileHandleEndpoint,
+    )
+
+
+async def create_access_requirements_if_none(
+    entity_id: str, synapse_client: Optional["Synapse"] = None
+) -> None:
+    """
+    Checks to see if the given entity has access requirements. If not, then one is added
+
+    Arguments:
+        entity_id: The ID of the entity.
+        synapse_client: If not passed in or None this will use the last client from
+            the `.login()` method.
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+    existing_restrictions = await client.rest_get_async(
+        f"/entity/{entity_id}/accessRequirement?offset=0&limit=1"
+    )
+    if len(existing_restrictions["results"]) <= 0:
```

Comment:
Can this ever be negative?

Nit: you can do this - `if not existing_restrictions["results"]`, but it's fine to be explicit as well.

Reply:
I hadn't stopped to consider this line. I took this from the `client.py` method:
```
    @tracer.start_as_current_span("Synapse::_createAccessRequirementIfNone")
    def _createAccessRequirementIfNone(self, entity: Union[Entity, str]) -> None:
        """
        Checks to see if the given entity has access requirements. If not, then one is added

        Arguments:
            entity: A Synapse ID or a Synapse Entity object
        """
        existingRestrictions = self.restGET(
            "/entity/%s/accessRequirement?offset=0&limit=1" % id_of(entity)
        )
        if len(existingRestrictions["results"]) <= 0:
            self.restPOST("/entity/%s/lockAccessRequirement" % id_of(entity), body="")
```

Looking at it the logic I can probably change this to:
`if existing_restrictions is None or not hasattr(existing_restrictions, "results") or len(existing_restrictions["results"]) == 0:`